error: writing 1 byte into a region of size 0

Compiling with gcc with Debug configuration works fine but Release throws the following:

```
/home/david/dev/heartcore/vendor/rapidjson/include/rapidjson/uri.h:330: error: writing 1 byte into a region of size 0 [-Werror=stringop-overflow=]
In file included from /home/david/dev/heartcore/vendor/rapidjson/include/rapidjson/pointer.h:19,
                 from /home/david/dev/heartcore/vendor/rapidjson/include/rapidjson/schema.h:19,
                 from /home/david/dev/heartcore/vendor/rapidjson/example/schemavalidator/schemavalidator.cpp:9:
/home/david/dev/heartcore/vendor/rapidjson/include/rapidjson/uri.h: In member function ‘void rapidjson::GenericUri<ValueType, Allocator>::Parse(const Ch*, std::size_t) [with ValueType = rapidjson::GenericValue<rapidjson::UTF8<> >; Allocator = rapidjson::CrtAllocator]’:
/home/david/dev/heartcore/vendor/rapidjson/include/rapidjson/uri.h:330:17: error: writing 1 byte into a region of size 0 [-Werror=stringop-overflow=]
  330 |         *query_ = '\0';
      |         ~~~~~~~~^~~~~~
```

--- a/var/spack/repos/builtin/packages/rapidjson/package.py
+++ b/var/spack/repos/builtin/packages/rapidjson/package.py
@@ -13,6 +13,11 @@ class Rapidjson(CMakePackage):
     url = "https://github.com/Tencent/rapidjson/archive/v1.1.0.tar.gz"
 
     version(
+        "1.2.0-2022-03-09",
+        git="https://github.com/Tencent/rapidjson.git",
+        commit="8261c1ddf43f10de00fd8c9a67811d1486b2c784"
+    )
+    version(
         "1.2.0-2021-08-13",
         git="https://github.com/Tencent/rapidjson.git",
         commit="00dbcf2c6e03c47d6c399338b6de060c71356464",

